test/test_nn.py

```diff
@@ -4897,6 +4897,26 @@ def test_batchnorm_cudnn_half(self):
             self.assertEqual(cudnn_output, thnn_output)
             self.assertEqual(cudnn_input_grad, thnn_input_grad, atol=1e-3, rtol=0)
 
+    @unittest.skipIf(not TEST_CUDA, "CUDA unavailable")
+    def test_batchnorm_cudnn_bf16(self):
+        # THNN
+        input = torch.randint(1, 10, (2, 3, 2, 2), dtype=torch.bfloat16, device="cuda", requires_grad=True)
```

Comment:
How is this choosing which backend to dispatch to? Would we need a context manager or something else to prevent it from going to cuDNN to generate the reference?